### PR TITLE
Address issues with Tabulator embed_content and optimize row expansion

### PR DESCRIPTION
**Pull request by philippjfr**
- Do not expand all rows by default if embed_content=True
- Do not auto-reset expanded rows if embed_content
- Ensure expanded rows are rendered efficiently

Fixes https://github.com/holoviz/panel/issues/7361

### CODE REVIEW

File: panel/models/tabulator.ts [hunk 1]
```diff
@@ -407,10 +407,21 @@ export class DataTabulatorView extends HTMLBoxView {
       for (const row of this.tabulator.rowManager.getRows()) {
         if (row.cells.length > 0) {
           const index = row.data._index
-          const icon = this.model.expanded.includes(index) ? "▼" : "►"
+          let icon
+          if (this.model.expanded.includes(index)) {
+            icon = "▼"
+          } else {
+            icon = "►"
+          }
```
Comment (philippjfr) [inline panel/models/tabulator.ts:415]:
```suggestion
          const icon = this.model.expanded.includes(index) ? "▼" : "►"
```

**Comment by philippjfr** [comment]
Sorry, misread it, this does look correct.

File: panel/models/tabulator.ts [hunk 2]
```diff
@@ -816,33 +822,23 @@ export class DataTabulatorView extends HTMLBoxView {
...
-  renderChildren(all: boolean = true): void {
-    new Promise(async (resolve: any) => {
-      let new_children = await this.build_child_views()
-      if (all) {
-        new_children = this.child_views
-      }
-      resolve(new_children)
-    }).then((new_children) => {
+  renderChildren(): void {
+    new Promise(async () => {
```
Comment (hoxbro) [inline panel/models/tabulator.ts:826]:
```suggestion
    void new Promise(async () => {
```

Nit, Raises an eslint error.

File: panel/models/tabulator.ts [hunk 3]
```diff
@@ -903,7 +906,9 @@ export class DataTabulatorView extends HTMLBoxView {
         }
       }
     }
+    this._updating_expanded = true
     this.model.expanded = expanded
```
Comment (hoxbro) [inline panel/models/tabulator.ts:910]: Maybe wrap this in a `try/finally`?

**Review by hoxbro — APPROVED** [review]
LGTM, I have added some nits comments.

File: panel/models/tabulator.ts (hunk 3 above)
Comment (philippjfr) [reply panel/models/tabulator.ts:910]: Will probably finally write a decorator of some kind for this.

**Comment by hoxbro** [comment]
pre-commit.ci autofix

File: panel/models/tabulator.ts [hunk 4]
```diff
@@ -852,33 +848,42 @@ export class DataTabulatorView extends HTMLBoxView {
...
+        this.resize_table()
+      }
+    }, () => this.root.has_finished())
+  }
+
+  resize_table(): void {
+    if (this.tabulator.rowManager.renderer != null) {
+      try {
+        this.tabulator.rowManager.adjustTableSize()
+      } catch() {}
```
Comment (philippjfr) [inline panel/models/tabulator.ts:884]:
```suggestion
      } catch(e) {}
```

File: panel/models/tabulator.ts [hunk 5]
```diff
@@ -852,33 +848,42 @@ export class DataTabulatorView extends HTMLBoxView {
...
+        this.resize_table()
+      }
+    }, () => this.root.has_finished())
+  }
+
+  resize_table(): void {
+    if (this.tabulator.rowManager.renderer != null) {
+      try {
+        this.tabulator.rowManager.adjustTableSize()
+      } catch(e) {}
```
Comment (philippjfr) [inline panel/models/tabulator.ts:884]:
```suggestion
      } catch (e) {}
```

**Pull request merged**